spikeinterface/core/addtemplatesrecording.py

```diff
@@ -44,12 +53,25 @@ def __init__(self, sorting: BaseSorting, templates: np.ndarray, nbefore: Union[L
         else:
             assert len(nbefore) == n_units
 
-        if amplitude_factor is None:
-            amplitude_factor = [[1.0]*sum([len(sorting.get_unit_spike_train(unit_id, segment_index=seg_index)) for seg_index in range(sorting.get_num_segments())]) for unit_id in sorting.unit_ids]
+        if type(amplitude_factor) is float:
```

Comment:
```suggestion
        if isinstance(amplitude_factor, float):
```
will catch both `float` and `np.float`

Reply:
Ah nice, thanks for the tip!